packages/contracts/src/viemClient.ts

```diff
@@ -0,0 +1,131 @@
+import type { ChainId } from '@shapeshiftoss/caip'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import assert from 'assert'
+import type { PublicClient } from 'viem'
+import { createPublicClient, fallback, http } from 'viem'
+import {
+  arbitrum,
+  arbitrumNova,
+  avalanche,
+  base,
+  bsc,
+  gnosis,
+  mainnet,
+  optimism,
+  polygon,
+} from 'viem/chains'
+
+export const viemEthMainnetClient = createPublicClient({
+  chain: mainnet,
+  transport: fallback([
+    http(process.env.REACT_APP_ETHEREUM_NODE_URL),
```

Comment:
we have the fallback, but this no longer explicitely checks for the env vars being defined and will use the default url provided by viem if undefined (which will most likely fail anyways due to csp)

Reply:
Ahh nice catch, 654ebe9659afba8d7d017c9653b5868a03346c82

Reply:
update to do the boolean filter on the urls vs the client (undefined url will use default client url as specified by viem): https://github.com/shapeshift/web/pull/7612/commits/a68ccc546ee88a1080633a7f11bfc51d420f1164